### PR TITLE
Add more TODOs requested by pull request reviewers

### PR DESCRIPTION
**Pull request by tuxji**
- Remove dependency on argp and gcc
- Report dev.dirs hanging problem
- Improve TDML Runner (actually my idea)

### CODE REVIEW

File: site/dev/design-notes/runtime2-todos.adoc [hunk 1]
```diff
@@ -36,6 +36,36 @@ If someone wants to help
 please let the mailto:dev@daffodil.apache.org[dev] list know
 in order to avoid duplication.
 
+=== Remove dependence on argp and gcc
+
+Update BUILD.md and the github actions to use clang instead of gcc.
```
Comment (mbeckerle) [inline site/dev/design-notes/runtime2-todos.adoc:41]: If we merge the work to master, then this TODO becomes a blocker-level bug, and needs a JIRA ticket.

File: site/dev/design-notes/runtime2-todos.adoc [hunk 2]
```diff
@@ -36,6 +36,36 @@ If someone wants to help
...
+and set CC=clang-10 and AR=llvm-ar-10 environment variables).
+
+Likewise, replace argp calls with getopt calls
+in daffodil_main.c and daffodil_argp.c
+even though it means giving up long options
+and other nice features that argp has.
+
+=== Report hanging problem running sbt (really dev.dirs) from MSYS2 on Windows
+
+We need to open a issue with a reproducible test case
```
Comment (mbeckerle) [inline site/dev/design-notes/runtime2-todos.adoc:53]: This feels like a JIRA ticket, with DAFFODIL-NNNN reference from the site where this hack fix lives.

Doesn't feel like it belongs in this TODO list. Too much of a "just an infrastructure bug" issue for that. 

Reminds me that we need to add a component to JIRA for runtime2 related tickets.

**Review by mbeckerle — APPROVED** [review]
+1 with suggestions on where to go with this info after runtime2 merge.

File: site/dev/design-notes/runtime2-todos.adoc (hunk 1 above)
Comment (tuxji) [reply site/dev/design-notes/runtime2-todos.adoc:41]: I've added [DAFFODIL-2500](https://issues.apache.org/jira/browse/DAFFODIL-2500) as a blocker-level issue.

File: site/dev/design-notes/runtime2-todos.adoc (hunk 2 above)
Comment (tuxji) [reply site/dev/design-notes/runtime2-todos.adoc:53]: Likewise, I've added [DAFFODIL-2501](https://issues.apache.org/jira/browse/DAFFODIL-2501) although I'll remove this todo later.

**Pull request merged**